ext/vulnsrc/ubuntu/ubuntu.go

```diff
@@ -13,369 +13,406 @@
 // limitations under the License.
 
 // Package ubuntu implements a vulnerability source updater using the
-// Ubuntu CVE Tracker.
+// Ubuntu Linux OVAL Database.
 package ubuntu
 
 import (
 	"bufio"
-	"errors"
+	"encoding/xml"
 	"fmt"
 	"io"
-	"os"
-	"path/filepath"
+	"net/http"
 	"regexp"
+	"strconv"
 	"strings"
+	"time"
+	"compress/bzip2"
 
 	log "github.com/sirupsen/logrus"
 
 	"github.com/coreos/clair/database"
 	"github.com/coreos/clair/ext/versionfmt"
 	"github.com/coreos/clair/ext/versionfmt/dpkg"
 	"github.com/coreos/clair/ext/vulnsrc"
-	"github.com/coreos/clair/pkg/gitutil"
+	"github.com/coreos/clair/pkg/commonerr"
 )
 
 const (
-	trackerURI   = "https://git.launchpad.net/ubuntu-cve-tracker"
-	updaterFlag  = "ubuntuUpdater"
-	cveURL       = "http://people.ubuntu.com/~ubuntu-security/cve/%s"
-	affectedType = database.SourcePackage
+	ovalURI = "https://people.canonical.com/~ubuntu-security/oval/"
+
+	// "Thu, 30 Nov 2017 03:07:57 GMT
+	timeFormatLastModified = "Mon, 2 Jan 2006 15:04:05 MST"
+
+	// timestamp format 2017-10-23T04:07:14
+	timeFormatOVAL = "2006-1-2T15:04:05"
+
+	updaterFlag = "ubuntuUpdater"
+
+	ubuntuOvalFilePrefix = "com.ubuntu."
 )
 
 var (
-	ubuntuIgnoredReleases = map[string]struct{}{
-		"upstream": {},
-		"devel":    {},
-
-		"dapper":   {},
-		"edgy":     {},
-		"feisty":   {},
-		"gutsy":    {},
-		"hardy":    {},
-		"intrepid": {},
-		"jaunty":   {},
-		"karmic":   {},
-		"lucid":    {},
-		"maverick": {},
-		"natty":    {},
-		"oneiric":  {},
-		"saucy":    {},
-
-		"vivid/ubuntu-core":          {},
-		"vivid/stable-phone-overlay": {},
-
-		// Syntax error
-		"Patches": {},
-		// Product
-		"product": {},
-	}
+	ignoredCriterions          []string
+	ubuntuPackageCommentRegexp = regexp.MustCompile(`^(.*) package in ([a-z]+) (?:(?:was vulnerable|is related to the CVE in some way) but has been fixed \(note: '(.*)'\)|is affected and needs fixing).$`)
+	ubuntuOvalFileRegexp       = regexp.MustCompile(`com.ubuntu.([a-z]+).cve.oval.xml.bz2`)
+)
+
+type oval struct {
```

Comment:
I don't have an issue with you defining the oval structure here, but you maybe interested in using our Oval parser fork: https://github.com/quay/goval-parser

Reply:
I just used the example set by the existing OVAL vulnsrc's.